test/service_test.cc

```diff
@@ -111,6 +113,44 @@ class ServiceTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
   int grpc_server_port_{0};
 };
 
+class ServiceTestWithParameterizedConstructor : public ServiceTest {
+public:
+  void SetUp() override {
+    auto logging_context = std::make_unique<Envoy::Logger::Context>(
+        spdlog::level::info, "%L %n [%g:%#] %v", log_lock_, false);
+    service_ = std::make_unique<ServiceImpl>(std::move(logging_context));
+    grpc::ServerBuilder builder;
+    loopback_address_ = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+
+    builder.AddListeningPort(fmt::format("{}:0", loopback_address_),
+                             grpc::InsecureServerCredentials(), &grpc_server_port_);
+    builder.RegisterService(service_.get());
+    server_ = builder.BuildAndStart();
+    setupGrpcClient();
+    setBasicRequestOptions();
+  }
+
+  Envoy::Thread::MutexBasicLockable log_lock_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ServiceTestWithParameterizedConstructor,
+                         ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                         Envoy::TestUtility::ipTestParamsToString);
+
+// Test single NH run with parameterized constructor
+TEST_P(ServiceTestWithParameterizedConstructor, BasicWithUserSpecifiedLoggingContext) {
+  auto r = stub_->ExecutionStream(&context_);
+  r->Write(request_, {});
+  r->WritesDone();
+  EXPECT_TRUE(r->Read(&response_));
+  ASSERT_TRUE(response_.has_error_detail());
+  EXPECT_THAT(response_.error_detail().message(), HasSubstr(std::string("Unknown failure")));
+  EXPECT_TRUE(response_.has_output());
+  EXPECT_GE(response_.output().results(0).counters().size(), 8);
+  auto status = r->Finish();
```

Comment:
auto critique applies here too, especially since there are two different statuses - rpc status and absl status, so it's important to distinguish which one is here.